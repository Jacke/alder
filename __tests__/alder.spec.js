const child_process = require('child_process')
const path = require('path')

const alder = path.resolve(__dirname, './../alder.js')


const execute = (args, cb) => {
  child_process.exec(`${alder} fixtures ${args}`, {}, (err, stdout, stderr) => {
    if (err) {
      throw err
    }
    expect(stdout).toMatchSnapshot()
    typeof cb !== 'undefined' && cb()
  })
}

describe('alder', () => {
  it(`works with 0 arguments/flags`, (done) => {
    execute(``, done)
  })

  it(`only renders the tree to a depth of 1`, (done) => {
    execute(`--depth 1`, done) && execute(`-d 1`, done)
  })

  it(`only prints directories`, (done) => {
    execute(`-D`, done) && execute(`--directories`, done)
  })

  it(`excludes files matching pattern (bar)`, (done) => {
    execute(`-e bar`, done) && execute(`--exclude bar`, done)
  })

  it(`prints the full path prefix for each file`, (done) => {
    execute(`-f`, done) && execute(`--full`, done)
  })

  it(`will not print indentation lines`, (done) => {
    execute(`-i`, done) && execute(`--no-indent`, done)
  })

  it(`ignores files listed in .gitignore (foo folder)`, (done) => {
    execute(`-I`, done) && execute(`--git-ignore`, done)
  })

  it(`will only show files matching pattern (foo)`, (done) => {
    execute(`-p foo`, done) && execute(`--include foo`, done)
  })

  it(`shows file size in tree`, (done) => {
    execute(`-s`, done) && execute(`--sizes`, done)
  })

  it(`prints the last modified date for each file`, (done) => {
    execute(`-t`, done) && execute(`--time-stamp`, done)
  })

  it(`prunes empty directories from output`, (done) => {
    execute(`--prune`, done)
  })

  it(`only descends directories with <2 files`, (done) => {
    execute(`--filelimit 2`, done)
  })

  it(`makes everything all jsxexy`, (done) => {
    execute(`--jsx`, done)
  })
})
