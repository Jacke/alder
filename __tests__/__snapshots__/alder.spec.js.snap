exports[`alder works with the  flag 1`] = `
"
 fixtures
 ├── README.md
 ├── bar
 │   ├── bar.js
 │   ├── baz.txt
 │   └── foo.js
 ├── baz
 │   └── baz.js
 ├── foo
 │   ├── README.md
 │   ├── bar.js
 │   └── foo.js
 ├── foobar.java
 ├── foobarbaz.txt
 ├── qux
 └── zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the --filelimit=3 flag 1`] = `
"
 fixtures
 ├── README.md
 ├── bar
 │   ├── bar.js
 │   ├── baz.txt
 │   └── foo.js
 ├── baz
 │   └── baz.js
 ├── foo
 │   ├── README.md
 │   ├── bar.js
 │   └── foo.js
 ├── foobar.java
 ├── foobarbaz.txt
 ├── qux
 └── zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the --jsx flag 1`] = `
"
 fixtures
<file size=\"0 B\" name=\"README.md\" />
<dir \"170 B\" name=\"bar\">
    <file size=\"0 B\" name=\"bar.js\" />
    <file size=\"0 B\" name=\"baz.txt\" />
    <file size=\"0 B\" name=\"foo.js\" />
</dir>
<dir \"102 B\" name=\"baz\">
    <file size=\"3.59 kB\" name=\"baz.js\" />
</dir>
<dir \"170 B\" name=\"foo\">
    <file size=\"0 B\" name=\"README.md\" />
    <file size=\"0 B\" name=\"bar.js\" />
    <file size=\"0 B\" name=\"foo.js\" />
</dir>
<file size=\"0 B\" name=\"foobar.java\" />
<file size=\"0 B\" name=\"foobarbaz.txt\" />
<dir size=\"68 B\" name=\"qux\" />
<file size=\"0 B\" name=\"zulu.rs\" />
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the --prune flag 1`] = `
"
 fixtures
 ├── README.md
 ├── bar
 │   ├── bar.js
 │   ├── baz.txt
 │   └── foo.js
 ├── baz
 │   └── baz.js
 ├── foo
 │   ├── README.md
 │   ├── bar.js
 │   └── foo.js
 ├── foobar.java
 ├── foobarbaz.txt
 └── zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -D flag 1`] = `
"
 fixtures
 ├── [Jan 23 12:41]  README.md
 ├── [Jan 23 12:40]  bar
 │   ├── [Jan 23 10:26]  bar.js
 │   ├── [Jan 23 12:40]  baz.txt
 │   └── [Jan 23 12:40]  foo.js
 ├── [Jan 23 10:26]  baz
 │   └── [Jan 23 04:45]  baz.js
 ├── [Jan 23 12:41]  foo
 │   ├── [Jan 23 12:41]  README.md
 │   ├── [Jan 23 12:40]  bar.js
 │   └── [Jan 23 10:26]  foo.js
 ├── [Jan 23 12:43]  foobar.java
 ├── [Jan 23 12:42]  foobarbaz.txt
 ├── [Jan 24 09:15]  qux
 └── [Jan 23 12:44]  zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -a flag 1`] = `
"
 fixtures
 ├── README.md
 ├── bar
 │   ├── bar.js
 │   ├── baz.txt
 │   └── foo.js
 ├── baz
 │   └── baz.js
 ├── foo
 │   ├── README.md
 │   ├── bar.js
 │   └── foo.js
 ├── foobar.java
 ├── foobarbaz.txt
 ├── qux
 └── zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -d flag 1`] = `
"
 fixtures
 ├── bar
 ├── baz
 ├── foo
 ├── qux
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -d flag 2`] = `
"
 fixtures
 ├── bar
 ├── baz
 ├── foo
 ├── qux
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -e bar flag 1`] = `
"
 fixtures
 ├── README.md
 ├── baz
 │   └── baz.js
 ├── foo
 │   ├── README.md
 │   └── foo.js
 ├── qux
 └── zulu.rs
4 directories, 8 files (3.59 kB)
"
`;

exports[`alder works with the -f flag 1`] = `
"
 fixtures
 ├── fixtures/README.md
 ├── fixtures/bar
 │   ├── fixtures/bar/bar.js
 │   ├── fixtures/bar/baz.txt
 │   └── fixtures/bar/foo.js
 ├── fixtures/baz
 │   └── fixtures/baz/baz.js
 ├── fixtures/foo
 │   ├── fixtures/foo/README.md
 │   ├── fixtures/foo/bar.js
 │   └── fixtures/foo/foo.js
 ├── fixtures/foobar.java
 ├── fixtures/foobarbaz.txt
 ├── fixtures/qux
 └── fixtures/zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -i flag 1`] = `
"
 fixtures
README.md
bar
bar.js
baz.txt
foo.js
baz
baz.js
foo
README.md
bar.js
foo.js
foobar.java
foobarbaz.txt
qux
zulu.rs
4 directories, 11 files (3.59 kB)
"
`;

exports[`alder works with the -p bar flag 1`] = `
"
 fixtures
 ├── bar
 │   ├── bar.js
 ├── foobar.java
 ├── foobarbaz.txt
4 directories, 7 files (0 B)
"
`;

exports[`alder works with the -s flag 1`] = `
"
 fixtures
 ├── README.md (0 B)
 ├── bar
 │   ├── bar.js (0 B)
 │   ├── baz.txt (0 B)
 │   └── foo.js (0 B)
 ├── baz
 │   └── baz.js (3.59 kB)
 ├── foo
 │   ├── README.md (0 B)
 │   ├── bar.js (0 B)
 │   └── foo.js (0 B)
 ├── foobar.java (0 B)
 ├── foobarbaz.txt (0 B)
 ├── qux
 └── zulu.rs (0 B)
4 directories, 11 files (3.59 kB)
"
`;
